resources/osl_mongodb.rb

```diff
@@ -0,0 +1,65 @@
+resource_name :osl_mongodb
+provides :osl_mongodb, platform_family: 'rhel'
+unified_mode true
+
+default_action :install
+
+property :version, String, name_property: true
+property :install_selinux_policy, [true, false], default: true
+property :data_dir, String, default: '/var/lib/mongo'
+property :log_dest, %w(syslog file), default: 'file'
+property :log_path, String, default: '/var/log/mongodb/mongod.log'
+property :port, Integer, default: 27017
+property :bind_ip, String, default: '127.0.0.1'
+property :max_connections, Integer, default: 51200
+
+action :install do
+  yum_repository 'mongodb-org' do
+    baseurl "https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/#{new_resource.version}/$basearch/"
+    repo_gpgcheck true
+    gpgkey "https://www.mongodb.org/static/pgp/server-#{new_resource.version}.asc"
+  end
+
+  package 'mongodb-org'
+
+  if new_resource.install_selinux_policy
+    selinux_install 'selinux'
+  end
+
+  template '/etc/mongod.conf' do
+    source 'mongod.conf.erb'
+    cookbook 'osl-resources'
+    owner 'mongod'
+    group 'mongod'
+    mode '0644'
+    variables(
+      data_dir: new_resource.data_dir,
+      log_dest: new_resource.log_dest,
+      log_path: new_resource.log_path,
+      port: new_resource.port,
+      bind_ip: new_resource.bind_ip,
+      max_connections: new_resource.max_connections
+    )
+    notifies :restart, 'service[mongod]', :immediately
+  end
+
+  sysctl 'vm.max_map_count' do
+    value "#{new_resource.max_connections * 2}"
+  end
```

Comment:
Where was this mentioned on the upstream document that it needed to be setup?

Reply:
I've tested this again and it looks like the warning is not showing up anymore, either in the mongodb logs or journalctl! I'll remove this in the next commit.

Reply:
Ok, removed and tested!